test/fail_compilation/bug9631.d

```diff
@@ -58,3 +58,54 @@ void test3()
     S[1] sa = cast(S[1])ta;
     auto t2 = cast(tem!().S[])sa;
 }
+
+/*
+TEST_OUTPUT:
+---
+fail_compilation/bug9631.d(79): Error: function `bug9631.arg.f(int i, S s)` is not callable using argument types `(int, S)`
+fail_compilation/bug9631.d(79):        cannot pass argument `y` of type `bug9631.tem!().S` to parameter `bug9631.S s`
```

Comment:
Note that line 66 calls it `bug9631.S` but line 65 calls it just `S`. Perhaps it would be better to just call it `S` on line 66, too.

Reply:
The point is to disambiguate between S and tem!().S. You're right that the module name isn't needed, but it makes it clear that the compiler has fully qualified the type. In the first line, there's no qualification so that it's easier to read the function parameters. Given this, if I remove the module name, it might not be so clear.

Reply:
ok